packages/internal/dex/sdk/src/exchange.ts

```diff
@@ -143,6 +145,133 @@ export class Exchange {
     };
   }
 
+  private async getUnwrapTransaction(
+    fromAddress: string,
+    tokenAmount: CoinAmount<ERC20>,
+    wimxInterface: ethers.utils.Interface,
+    gasPrice: CoinAmount<Native> | null,
+  ) : Promise<WrapUnwrapTransactionDetails> {
+    const calldata = wimxInterface.encodeFunctionData('withdraw', [tokenAmount.value]);
+    // 35216 is the upper bound of the gas estimate for the withdraw function.
+    // The differenfce between upper and normal is less than 0.0001 IMX, so we just use the upper bound
+    // to avoid an extra remote call.
+    const gasEstimate = ethers.BigNumber.from('35216');
+
+    const approval = await getApproval(
+      this.provider,
+      fromAddress,
+      {
+        spender: this.wrappedNativeToken.address, amount: tokenAmount,
+      },
+      gasPrice,
+    );
+
+    const gasFeeEstimate = gasPrice ? toPublicAmount(calculateGasFee(gasPrice, gasEstimate)) : null;
+    // 3. Get the transaction details. This is calling `deposit` or `withdraw` on the WETH/WIMX contract.
+    const transactionDetails: TransactionDetails = {
+      transaction: {
+        data: calldata,
+        to: this.wrappedNativeToken.address, // wrapping and unwrapping is done on the WETH/WIMX contract itself
+        from: fromAddress,
+      },
+      gasFeeEstimate,
+    };
+
+    return {
+      transaction: transactionDetails,
+      approval,
+    };
+  }
+
+  private getWrapTransaction(
+    fromAddress: string,
+    tokenAmount: CoinAmount<ERC20>,
+    wimxInterface: ethers.utils.Interface,
+    gasPrice: CoinAmount<Native> | null,
+  ) : WrapUnwrapTransactionDetails {
+    const calldata = wimxInterface.encodeFunctionData('deposit');
+    // 45038 is the upper bound of the gas estimate for the deposit function.
+    // This upper bound is reached when a user is wrapping for the first time (cold + unused storage slot)
+    // The lower bound is 27,938, so the difference is 17,100 gas. If transactions are 10 gwei IMX,
+    // this is about 0.000171 IMX difference. In order to avoid an extra remote call, we just use the upper bound instead.
+    const gasEstimate = ethers.BigNumber.from('45038');
+
+    const gasFeeEstimate = gasPrice ? toPublicAmount(calculateGasFee(gasPrice, gasEstimate)) : null;
+    // Get the transaction details. This is calling `deposit` or `withdraw` on the WETH/WIMX contract.
+    const transactionDetails: TransactionDetails = {
+      transaction: {
+        data: calldata,
+        to: this.wrappedNativeToken.address, // wrapping and unwrapping is done on the WETH/WIMX contract itself
+        value: ethers.BigNumber.from(tokenAmount.value), // Wrapping involves sending the native asset as TX.value
```

Comment:
`tokenAmount.value` is a bignumber already so shouldn't need to use `.from` here

Reply:
Good catch ty